src/diffusers/models/unet_2d_condition.py

```diff
@@ -135,7 +135,9 @@ def forward(
         if not torch.is_tensor(timesteps):
             timesteps = torch.tensor([timesteps], dtype=torch.long, device=sample.device)
         elif torch.is_tensor(timesteps) and len(timesteps.shape) == 0:
-            timesteps = timesteps[None].to(sample.device)
+            if sample.device.type == "mps":
+                timesteps = timesteps.to(dtype=torch.float32)
```

Comment:
shouldn't this be handled in the `if` block also ?

also can we always cast it to `float32` cc @patrickvonplaten ?

Reply:
I think I debugged the other branch and it worked fine (at least when using an `int`), I'll double check to make sure.

Reply:
Yes I think we can always cast here indeed